lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1216,10 +1219,18 @@ for (my $i=1 ; $i <= $request->{overpayment_qty}; $i++) {
      # Now we split the account selected options, using the namespace the if statement
      # provides for us.
      $request->{"overpayment_topay_$i"} = LedgerSMB::PGNumber->from_input($request->{"overpayment_topay_$i"});
-     $request->{"overpayment_account_$i"} =~ /^(\d+)--*/;
-     my $id = $1;
-     $request->{"overpayment_cash_account_$i"} =~ /^(\d+)--*/;
-     my $cashid = $1;
+     my $id;
+     if ( $request->{"overpayment_account_$i"} =~ /^(\d+)--*/) {
+        $id = $1;
+     } else {
+        die "Error: Invalid data";
```

Comment:
Overpayment accounts aren't required to be set up on a company either. They are defined in the database in the `account` table, with referencing records in the `account_link` table where the `description` equals either `AR_overpayment` or `AP_overpayment`.